### PR TITLE
Make AR1 closer to the original implementation

### PR DESCRIPTION
**Pull request by BenCrulis**
This PR fixes AR1 freezing the whole model by default and allows to choose the incremental learning rate, making the implementation closer to the original at https://github.com/vlomonaco/ar1-pytorch

With these fixes I am able to nearly replicate the results of the original github on CORe50.

### CODE REVIEW

**Comment by AntonioCarta** [comment]
@lrzpellegrini it seems that checkpointing tests are failing here. Any idea why?

**Comment by BenCrulis** [comment]
@AntonioCarta probably because of the added parameter (`inc_lr`) that isn't in the pickled objects I would say, though I'm not sure.

**Comment by AntonioCarta** [comment]
After discussing this with @lrzpellegrini, it appears that some changes in Avalanche dependencies broke the checkpointing. Now, since the errors are unrelated to this PR (you can see python 3.10 is working) I'm going to merge this anyway. @lrzpellegrini is working to fix checkpointing.
